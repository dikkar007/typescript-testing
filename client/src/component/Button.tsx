import * as React from 'react';

class Button extends React.Component {
  public render() {
    return (
      <div className="button">{this.props.children}</div>
    );
  }
}

export default Button;
