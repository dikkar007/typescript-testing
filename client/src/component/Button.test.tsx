import * as React from 'react';
import * as TestUtils from 'react-dom/test-utils';
import Button from './Button';

it('Button renders correctly', () => {
  const tree = TestUtils.renderIntoDocument(<Button>hi!</Button>);
  expect(tree).toMatchSnapshot();
});

