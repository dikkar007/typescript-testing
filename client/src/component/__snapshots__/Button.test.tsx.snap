// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
Button {
  "_reactInternalFiber": FiberNode {
    "_debugID": 4,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": null,
    "actualDuration": 0,
    "actualStartTime": 0,
    "alternate": null,
    "child": FiberNode {
      "_debugID": 5,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": 0,
      "alternate": null,
      "child": null,
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": "hi!",
        "className": "button",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": "hi!",
        "className": "button",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseTime": 0,
      "sibling": null,
      "stateNode": <div
        class="button"
      >
        hi!
      </div>,
      "tag": 5,
      "treeBaseTime": 0,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "children": "hi!",
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "children": "hi!",
    },
    "ref": null,
    "return": FiberNode {
      "_debugID": 1,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": 0,
      "alternate": FiberNode {
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": 0,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseTime": 0,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
              class="button"
            >
              hi!
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "earliestPendingTime": 0,
          "earliestSuspendedTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "latestPendingTime": 0,
          "latestPingedTime": 0,
          "latestSuspendedTime": 0,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "treeBaseTime": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "expirationTime": 1,
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": Object {
            "callback": null,
            "expirationTime": 1,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <Button>
                hi!
              </Button>,
            },
            "tag": 0,
          },
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": Object {
            "callback": null,
            "expirationTime": 1,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <Button>
                hi!
              </Button>,
            },
            "tag": 0,
          },
        },
      },
      "child": [Circular],
      "effectTag": 32,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Button>
          hi!
        </Button>,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseTime": 0,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
            class="button"
          >
            hi!
          </div>
        </div>,
        "context": Object {},
        "current": [Circular],
        "earliestPendingTime": 0,
        "earliestSuspendedTime": 0,
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "latestPendingTime": 0,
        "latestPingedTime": 0,
        "latestSuspendedTime": 0,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "treeBaseTime": 0,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <Button>
            hi!
          </Button>,
        },
        "expirationTime": 0,
        "firstCapturedEffect": null,
        "firstCapturedUpdate": null,
        "firstEffect": null,
        "firstUpdate": null,
        "lastCapturedEffect": null,
        "lastCapturedUpdate": null,
        "lastEffect": null,
        "lastUpdate": null,
      },
    },
    "selfBaseTime": 0,
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "treeBaseTime": 0,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "children": "hi!",
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
